### DISCUSSION TITLE
Any chance to use the Calling SDK in Angular 18+?

### DISCUSSION THREAD

**Discussion by seimic** [webex/webex-js-sdk#3840] [General]
Hi,

did anyone try to use the Calling SDK in Angular and succeeded?
If I look in the code from the example in [https://webex.github.io/webex-js-sdk/samples/calling/](https://webex.github.io/webex-js-sdk/samples/calling/)
(Code at: [https://github.com/webex/webex-js-sdk/blob/next/docs/samples/calling/app.js](https://github.com/webex/webex-js-sdk/blob/next/docs/samples/calling/app.js)) things seem
to be clear and relatively simple to set up. 

Startet a new Angular 18 project with electron to get all the dependencies from Node into the web container
but I can't get the initialization to work. At runtime there is no such method like [Calling.init(...)](https://github.com/webex/webex-js-sdk/blob/11f97f5b80888b8c6eb6aea3f72b33c14d620bc2/docs/samples/calling/app.js#L287C3-L287C62) as mentioned here
[https://developer.webex.com/docs/sdks/webex-calling-sdk-web-quickstart](https://developer.webex.com/docs/sdks/webex-calling-sdk-web-quickstart). Tried the older version too [https://github.com/webex/webex-js-sdk/wiki/Quickstart-Guide-(Calling)#calling-configuration](https://github.com/webex/webex-js-sdk/wiki/Quickstart-Guide-(Calling)#calling-configuration) but don't get
why it doesn't work.

Does anyone have a typings file for the calling API? I mean the calling.min.js for example.

Kind regards,
Michael

**Comment by sreenara** [comment]
@seimic sorry about the delay, we didn't look at the discussions for some time. The `Calling` object won't be present if you directly consume `@webex/calling`. Please refer to this [Quickstart guide](https://developer.webex.com/docs/sdks/webex-calling-sdk-web-quickstart) to see how to import and use the package. You'll have to try the `createClient` method.

**Reply by seimic** [reply]
OK thanks. Finally I used only the events, webhooks and few commands to place a call, receive a call (from the /telephony/calls API) etc. to integrate it with Webex Desktop. It works fine without the need to use the Calling API fully integrated in the website (a company internal website).

**Reply by sreenara** [reply]
That's good to hear. Thanks for the update @seimic
